static/src/javascripts/projects/common/modules/support/articleCount.ts

```diff
@@ -0,0 +1,87 @@
+import { storage } from '@guardian/libs';
+import {
+	getWeeklyArticleHistory,
+	incrementWeeklyArticleCount,
+} from '@guardian/support-dotcom-components';
+import type { WeeklyArticleHistory } from '@guardian/support-dotcom-components/dist/shared/src/types/targeting';
+import { storageKeyDailyArticleCount } from 'common/modules/onward/history';
+import { getArticleCountConsent } from 'common/modules/support/supportMessaging';
+
+export interface DailyArticle {
+	day: number;
+	count: number;
+}
+
+type DailyArticleHistory = DailyArticle[];
+
+const today = Math.floor(Date.now() / 86400000); // 1 day in ms
+
+const getDailyArticleHistory = (): DailyArticleHistory | undefined => {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- reading from local storage
+	const item = storage.local.get(storageKeyDailyArticleCount);
+	if (item) {
+		return item as DailyArticleHistory;
+	}
+	return undefined;
```

Comment:
Not sure if this is necessary as not returning something returns `undefined`.

Reply:
true - it felt more deliberate/explicit, but I can remove